cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -1067,8 +1067,38 @@ def make_resource(self, data, **kwargs):
         return QueueImage(**data)
 
 
-class HeadNodeCustomActionSchema(BaseSchema):
-    """Represent the schema of the custom action."""
+class OneOrManyCustomActionField(fields.Nested):
+    """Custom Marshmallow filed to handle backward compatible single script custom actions."""
+
+    def __init__(self, **kwargs):
+        schema = (
+            UpdatableCustomActionScriptSchema
+            if kwargs.get("metadata", {}).get("update_policy") == UpdatePolicy.SUPPORTED
+            else CustomActionScriptSchema
+        )
+        super().__init__(schema, **kwargs)
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        if "Script" in value and "Sequence" in value:
+            raise ValidationError("Both Script and Sequence fields are provided. Only one is allowed.")
+
+        if "Script" in value:
+            return super()._deserialize(value, attr, data, **kwargs)
+
+        if "Sequence" in value:
+            sequence = value["Sequence"]
+            if not isinstance(sequence, list):
+                raise ValidationError("Invalid input type for Sequence, expected list.")
```

Comment:
question: non-blocking: Adding a unit test for this should be quite easy, why don't you do that?

Reply:
adding it 😊